### ISSUE TITLE
Async connection close method doesn't terminate the ongoing transaction

### ISSUE THREAD

**Issue by khushboovashi** [psycopg/psycopg#608]
If the async query is running in the thread and the connection is explicitly closed, the ongoing execution doesn't get terminated (we can still see the process in the terminal). Also, the execute function ( await _c.execute("SELECT pg_sleep(20);")) runs indefinitely.

Here is the test code:

```
async def test_connect():
    conn = await AsyncConnection.connect(
        host='localhost',
        port=5432,
        dbname='db1',
        user='postgres',
        password='xxxxxx'
    )

    return conn

conn = asyncio.run(test_connect())
_cur = _cursor(conn)

def asyn_exec_query(c):
    try:
        async def _execute(_c):
            print("Execution Start.")
            await _c.execute("SELECT pg_sleep(20);")
            print("Execution Complete.")
        asyncio.run(_execute(c))
    except Exception as e:
        print(str(e))


async def main():
    _thread = threading.Thread(target=asyn_exec_query, args=(_cur,))
    _thread.start()
    print("Before Sleep")
    time.sleep(3)
    print("After sleep")
    await conn.close()
    print("Finish")

asyncio.run(main())
```

**Comment by dvarrazzo** [comment]
Uhmm... so the suggestion would be to send a cancel before closing, if the connection is closed in ACTIVE state?

**Comment by khushboovashi** [comment]
Yes, and the cursor.execute function should not go into an infinite loop.

**Comment by dvarrazzo** [comment]
This should be fixed in 3.1.12, just released.
